sunpy/time/time.py

```diff
@@ -126,83 +129,71 @@ def _astropy_time(time):
     return time if isinstance(time, astropy.time.Time) else astropy.time.Time(parse_time(time))
 
 
-def _parse_dt64(dt):
-    """
-    Parse a single numpy datetime64 object
-    """
-    # Convert to microsecond precision because datetime cannot handle nanoseconds
-    return dt.astype('M8[us]').astype(datetime)
-
-
 @singledispatch
-def convert_time(time_string, **kwargs):
+def convert_time(time_string, format=None, **kwargs):
     # default case when no type matches
-    raise ValueError("'{tstr!s}' is not a valid time string!".format(tstr=time_string))
-
-
-@convert_time.register(pandas.Timestamp)
-def convert_time_pandasTimestamp(time_string, **kwargs):
-    return time_string.to_pydatetime()
+    try:
+        return ap.Time(time_string, format=format, **kwargs)
+    except (ValueError, TypeError):
+        raise ValueError("'{tstr!s}' is not a valid time string!".format(tstr=time_string))
 
 
 @convert_time.register(pandas.Series)
 def convert_time_pandasSeries(time_string, **kwargs):
-    if 'datetime64' in str(time_string.dtype):
-        return np.array([dt.to_pydatetime() for dt in time_string])
-    else:
-        convert_time.dispatch(object)(time_string, **kwargs)
+    return ap.Time(time_string.tolist(), **kwargs)
 
 
 @convert_time.register(pandas.DatetimeIndex)
 def convert_time_pandasDatetimeIndex(time_string, **kwargs):
-    return time_string._mpl_repr()
+    return ap.Time(time_string.tolist(), **kwargs)
 
 
+@convert_time.register(pandas.Timestamp)
 @convert_time.register(datetime)
 def convert_time_datetime(time_string, **kwargs):
-    return time_string
+    return ap.Time(time_string, **kwargs)
 
 
 @convert_time.register(date)
 def convert_time_date(time_string, **kwargs):
-    return datetime.combine(time_string, time())
+    return ap.Time(time_string.isoformat(), **kwargs)
 
 
 @convert_time.register(tuple)
 def convert_time_tuple(time_string, **kwargs):
-    return datetime(*time_string)
+    return ap.Time('{}-{}-{}'.format(*time_string), **kwargs)
 
 
 @convert_time.register(float)
 @convert_time.register(int)
 def convert_time_float(time_string, **kwargs):
-    return datetime(1979, 1, 1) + timedelta(0, time_string)
+    return ap.Time(time_string, format='utime', **kwargs)
```

Comment:
I don't think we need this anymore. `utime` is not the only valid float input to `Time` and you have to specify `format='utime'` if you want to provide `parse_time` with a utime, so this should just be handled by the default case.

Reply:
A number of `Time` formats have float and int input. `jyear`, `jd`, `gps`, `decimalyear` etc.

Reply:
exactly, so registering this for all float inputs is wrong. We should just hand this off to the `Time` constructor.